### ISSUE TITLE
Shouldn't print datasouce user&&password when datasource crash

### ISSUE THREAD

**Issue by mkdym** [apache/superset#11822]
Superset prints clickhouse datasource user&&password within connection string when datasource crash.

### Expected results

Do not print user&&password.

### Actual results

Print user&&password.

#### Screenshots

![image](https://user-images.githubusercontent.com/10892624/100317090-b4665e00-2ff6-11eb-9177-852df9b70258.png)

#### How to reproduce the bug

1. Add clickhouse datasource
2. Make chart with clickhouse datasource table.
3. Shutdown clickhouse server.
4. Open above chart.

### Environment

- superset version: `0.37.2`
- python version: `python --version`
- docker image repo: `https://github.com/amancevice/docker-superset`
- docker image version: `amancevice/superset:0.37.2`

**Comment by dpgaspar** [comment]
Thank you for reporting, can you provide the server logs?

**Comment by mkdym** [comment]
![image](https://user-images.githubusercontent.com/10892624/100401206-c2b68780-3093-11eb-97f0-d47a1b43dcf4.png)

Reproduced by these steps:
1. deploy clickhouse-server(yandex/clickhouse-server:20.3) and superset(amancevice/superset:0.37.2)
2. add clickhouse databases to superset
3. open sqllab, select clickhouse table, do query 'SELECT now()'
4. docker stop clickhouse-server
5. click 'run query' again

I fixed the issue temporarily by filter reponse json message:
![image](https://user-images.githubusercontent.com/10892624/100402398-1aa2bd80-3097-11eb-8629-35f9c9cd9d37.png)
